tools/build.sh

```diff
@@ -1,26 +1,25 @@
 #!/bin/bash
-
 # Copyright 2018 SMF Authors
 #
 
-
 set -e
-
-
 if [[ ! -z ${CI} ]]; then
-    echo "In continous integration system..."
-    set -x
+  echo "In continous integration system..."
+  set -x
 fi
 
+this_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+rootdir="$(cd ${this_dir}/.. && pwd)"
+
+build_rootdir=${rootdir}/build
+builddir=""
+
 . /etc/os-release
 
-this_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-root="$(cd ${this_dir}/.. && pwd)"
-buildtype="debug"
-builddir=$root/build
 function buildcmd() {
-    ninja -C $builddir
+  make -j$(nproc) -C ${1}
```

Comment:
did you test doing a `cmake -GNinja ..` by chance ?

Reply:
tried on ubuntu:bionic and it took some work to get functioning but it seems ok now. haven't tried on other distros.